components/org.wso2.carbon.identity.application.authenticator.push.core/src/main/java/org/wso2/carbon/identity/application/authenticator/push/core/PushJWTValidator.java

```diff
@@ -0,0 +1,221 @@
+     * @param jwt   JWT string
+     * @param claim Name of the required claim
+     * @return Retrieved claim
+     */
+    public String getJWTClaim(String jwt, String claim) {
+
+        try {
+            if (!isJWT(jwt)) {
+                return null;
+            } else {
+                SignedJWT signedJWT = SignedJWT.parse(jwt);
+                JWTClaimsSet claimSet = signedJWT.getJWTClaimsSet();
+                if (claimSet != null) {
+                    return (String) claimSet.getClaim(claim);
+                } else {
+                    return null;
+                }
+            }
+        } catch (ParseException e) {
+            if (!log.isDebugEnabled()) {
+                log.error("Error occurred while parsing JWT string", e);
+            }
+            return null;
+        }
+    }
+
+    /**
+     * Get Device ID from JWT.
+     *
+     * @param jwt JWT string
+     * @return Device ID
+     */
+    public String getDeviceId(String jwt) {
+
+        return getJWTClaim(jwt, "did");
+    }
+
+    /**
+     * Get Session data key from JWT.
+     *
+     * @param jwt JWT string
+     * @return Session data key
+     */
+    public String getSessionDataKey(String jwt) {
+
+        return getJWTClaim(jwt, "sid");
+    }
+
+    /**
+     * Get Authentication status from JWT.
+     *
+     * @param jwt JWT string
+     * @return Authentication status
+     */
+    public String getAuthStatus(String jwt) {
+
+        return getJWTClaim(jwt, "res");
+    }
+
+    /**
+     * Validate if the JWT is expired.
+     *
+     * @param expirationTime Time set for the JWT to expire
+     * @return Boolean validating if the JWT is not expired
+     */
+    private boolean checkExpirationTime(Date expirationTime) {
+
+        long timeStampSkewMillis = OAuthServerConfiguration.getInstance().getTimeStampSkewInSeconds() * 1000;
+        long expirationTimeInMillis = expirationTime.getTime();
+        long currentTimeInMillis = System.currentTimeMillis();
+        return (currentTimeInMillis + timeStampSkewMillis) <= expirationTimeInMillis;
+    }
+
+    /**
+     * Validate if the JWT is active.
+     *
+     * @param notBeforeTime Time set to activate the JWT
+     * @return Boolean validating if the JWT is active
+     */
+    private boolean checkNotBeforeTime(Date notBeforeTime) {
+
+        if (notBeforeTime != null) {
+            long timeStampSkewMillis = OAuthServerConfiguration.getInstance().getTimeStampSkewInSeconds() * 1000;
+            long notBeforeTimeMillis = notBeforeTime.getTime();
+            long currentTimeInMillis = System.currentTimeMillis();
+            return currentTimeInMillis + timeStampSkewMillis >= notBeforeTimeMillis;
+
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Validate legitimacy of JWT.
+     *
+     * @param tokenIdentifier JWT string
+     * @return Boolean validation for if the string is a JWT
+     */
+    private boolean isJWT(String tokenIdentifier) {
```

Comment:
Do a validation similar to whats done [here](https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java#L2998)